samples/test/vision_object_detection_create_dataset.test.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const {assert} = require('chai');
+const {AutoMlClient} = require('@google-cloud/automl').v1;
+
+const cp = require('child_process');
+const uuid = require('uuid');
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+
+const CREATE_DATASET_REGION_TAG = 'vision_object_detection_create_dataset';
+const LOCATION = 'us-central1';
+
+describe('Automl Vision Object Detection Create Dataset Test', () => {
+  const client = new AutoMlClient();
+  let datasetId;
+
+  it('should create a dataset', async () => {
+    const projectId = await client.getProjectId();
+    const displayName = `test_${uuid
+      .v4()
```

Comment:
In the real world, it might be worth replacing `-` with an empty string, rather than `_`; then you'd keep 16 more bits of entropy. For tests, I think you're still sitting at `2^66` possible values, vs., `2^122` without the slice logic .... tldr; you're probably not going to collide anytime soon.